src/components/Button/Button.tsx

```diff
@@ -1,13 +1,80 @@
-import { Text, TouchableOpacity } from 'react-native'
+import React from 'react'
+import { MaterialIcons } from '@expo/vector-icons'
+import { View } from 'react-native'
+import {
+  Button as PaperButton,
+  ButtonProps as PaperButtonProps,
+} from 'react-native-paper'
 
-import { ButtonProps } from './Button.types'
+export interface ButtonProps
+  extends Omit<PaperButtonProps, 'icon' | 'children'> {
+  title?: string | React.ReactNode
+  onPress?: () => void
+  materialIconName?: keyof typeof MaterialIcons.glyphMap
+  icon?: React.ReactNode
+  alignIcon?: 'left' | 'right'
+  loading?: boolean
+}
```

Comment:
Pelo que entendi aceitamos ou um `materialIconName`, ou um `icon`, é isso? Se for o caso, podemos tentar refletir isso no tipo do Prop do botão. Vou dar um exemplo com `type` ao invés de `interface`, que é o que estou mais acosutmado, mas sinta-se a vontade pra converter de volta
```suggestion
type IconProps =
  | {
      materialIconName?: keyof typeof MaterialIcons.glyphMap
      icon: undefined
    }
  | {
      materialIconName: undefined
      icon?: React.ReactNode
    };

type ButtonProps = Omit<PaperButtonProps, "icon" | "children"> & {
  title?: string | React.ReactNode
  onPress?: () => void
  alignIcon?: "left" | "right"
  loading?: boolean
} & IconProps
```
Aliás, no meu exmelpo eu tirei o export do `ButtonProps`. Não verifiquei no resto do projeto se algum componente externo usa ele, se sim beleza deixar ele ali. Posso estar errado, mas eu imagino que ninguém use.

Reply:
Eu só trocaria o que o @MatheusBuss sugeriu pra

```ts
type IconProps =
	  | {
	      materialIconName?: keyof typeof MaterialIcons.glyphMap
	      icon?: undefined
	    }
	  | {
	      materialIconName?: undefined
	      icon?: React.ReactNode
	    };
```

pra não precisar passar `icon={icon} materialIconName={undefined}`, ou vice versa

Reply:
Farei isso, ficou muito bom!!